jmclient/test/test_yieldgenerator.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import (absolute_import, division,
+                        print_function, unicode_literals)
+from builtins import * # noqa: F401
+from future.utils import iteritems
+
+import unittest
+
+from jmclient import load_program_config, jm_single,\
+    SegwitLegacyWallet, VolatileStorage, YieldGeneratorBasic, get_network
+
+
+class TestWallet(SegwitLegacyWallet):
+    """A wallet instance that makes it easy to do the tasks we need for
+    the tests.  It has convenience methods to add UTXOs with pre-defined
+    balances for all mixdepths, and to verify if a given UTXO or address
+    corresponds to the mixdepth it should."""
+
+    def __init__(self, balances):
+        """Creates the wallet, setting the balances of the mixdepths
+        as given by the array.  (And the number of mixdepths from the array
+        elements."""
+
+        load_program_config()
+
+        storage = VolatileStorage()
+        super(TestWallet, self).initialize(storage, get_network(),
+                                           max_mixdepth=len(balances)-1)
+        super(TestWallet, self).__init__(storage)
+
+        for m, b in enumerate(balances):
+            self.add_utxo_at_mixdepth(m, b)
+
+    def add_utxo_at_mixdepth(self, mixdepth, balance):
+        tx = {'outs': [{'script': self.get_internal_script(mixdepth),
+                        'value': balance}]}
+        # We need to generate a fake "txid" that has to be unique for all the
+        # UTXOs that are added to the wallet.  For that, we simply use the
+        # script, and make it fit the required length (32 bytes).
+        txid = tx['outs'][0]['script'] + b'x' * 32
+        txid = txid[:32]
+        self.add_new_utxos_(tx, txid)
+
+    def assert_utxos_from_mixdepth(self, utxos, expected):
+        """Asserts that the list of UTXOs (as returned from UTXO selection
+        of the wallet) is all from the given expected mixdepth."""
+
+        for _, u in iteritems(utxos):
```

Comment:
minor nit, this should be `for u in utxos.values():`

Reply:
Ah yes, Python is not my main language.  Thanks for pointing this out!